app/views/Export/ExportCodeInput.js

```diff
@@ -0,0 +1,235 @@
+
+  const focus = i => {
+    digitRefs.current[i].focus();
+  };
+
+  // Focus on mount
+  useEffect(() => {
+    setTimeout(() => {
+      focus(0);
+    }, 0); // allow waiting for transition to end & first paint
+  }, []);
+
+  const onFocus = i => {
+    if (i > currentIndex) {
+      // prohibit skipping forward
+      focus(currentIndex);
+    } else {
+      // restart at clicked digit
+      setCurrentIndex(i);
+      setCode(code.slice(0, i));
+    }
+  };
+
+  // Adding digits
+  const onChangeDigit = d => {
+    if (d.length) {
+      setCode(code.slice(0, currentIndex) + d);
+      const nextIndex = currentIndex + 1;
+      if (nextIndex < length) {
+        setCurrentIndex(nextIndex);
+        focus(nextIndex);
+      }
+    }
+  };
+
+  // Removing digits
+  const onKeyPress = e => {
+    if (e.nativeEvent.key === 'Backspace') {
+      // go to previous
+      if (!code[currentIndex]) {
+        const newIndex = currentIndex - 1;
+        if (newIndex >= 0) {
+          setCurrentIndex(newIndex);
+          setCode(code.slice(0, newIndex));
+          focus(newIndex);
+        }
+      }
+      // clear current (used for last digit)
+      else {
+        setCode(code.slice(0, currentIndex));
+      }
+    }
+  };
+
+  return (
+    <View style={{ flexDirection: 'row', flexShrink: 1 }}>
+      {[...digits].map((digit, i) => (
+        <TextInput
+          ref={ref => (digitRefs.current[i] = ref)}
+          key={`${i}CodeDigit`}
+          value={digit}
+          style={[
+            styles.digitInput,
+            {
+              borderColor: digit
+                ? Colors.VIOLET_BUTTON
+                : Colors.VIOLET_BUTTON_LIGHT,
+            },
+          ]}
+          keyboardType={'number-pad'}
+          returnKeyType={'done'}
+          onChangeText={onChangeDigit}
+          onKeyPress={onKeyPress}
+          blurOnSubmit={false}
+          onFocus={() => onFocus(i)}
+        />
+      ))}
+    </View>
+  );
+};
+
+export const ExportSelectHA = ({ route, navigation }) => {
+  const { t } = useTranslation();
+
+  const [code, setCode] = useState('');
+  const [isCheckingCode, setIsCheckingCode] = useState(false);
+  const [codeInvalid, setCodeInvalid] = useState(false);
+
+  const { selectedAuthority } = route.params;
+
+  const validateCode = async () => {
+    setIsCheckingCode(true);
+    setCodeInvalid(false);
+    try {
+      const res = await fetch(`${MOCK_ENDPOINT}?access_code=${code}`);
+      const { valid } = await res.json();
+      if (valid) {
+        navigation.navigate('ExportLocationConsent', {
+          selectedAuthority,
+          code,
+        });
+      } else {
+        setCodeInvalid(true);
+      }
+      setIsCheckingCode(false);
+    } catch (e) {
+      Alert.alert(t('common.something_went_wrong'));
+      setIsCheckingCode(false);
+    }
+  };
+
+  return (
+    <Theme use='default'>
+      <StatusBar
+        barStyle='dark-content'
+        backgroundColor={Colors.VIOLET_BUTTON}
```

Comment:
You can get the theme bg color via `useTheme().background`

Reply:
It looks like the background mixes backgroundColor & gradient styled instructions?

I updated the color here to be consistent with the hard coded theming color of the background, for consistency on android with the background color